aten/src/ATen/TensorIterator.cpp

```diff
@@ -402,9 +402,11 @@ void TensorIteratorBase::compute_types(const TensorIteratorConfig& config) {
   }
 
   // Checks that either the computation type is computable or unneeded
-  TORCH_INTERNAL_ASSERT(!(has_different_input_dtypes && !config.promote_inputs_to_common_dtype_ &&
-                        (has_undefined_outputs || config.enforce_safe_casting_to_output_ ||
-                        config.cast_common_dtype_to_outputs_)));
+  TORCH_CHECK(
+    !(has_different_input_dtypes && !config.promote_inputs_to_common_dtype_ &&
+      (has_undefined_outputs || config.enforce_safe_casting_to_output_ ||
+       config.cast_common_dtype_to_outputs_)),
+    "Cannot promote types given these arguments");
```

Comment:
update: i'll change the message accordingly, but i don't see a way to print the operation name. i don't think there's a way to get this information from TI. or am i mistaken? so rn i'm only planning to print the operand dtypes as the op_name thing would require adding op_name everywhere where TI config is constructed

Reply:
i did this for now:

```cpp
  std::ostringstream input_dtypes;

  for (auto& op : operands_) {
    // Get input dtypes for error reporting.
    if (!op.is_output) {
      if (input_dtypes.tellp() == 0) {
        input_dtypes << op.current_dtype;
      } else {
        input_dtypes << ", " << op.current_dtype;
      }
    }
```

what i dislike about this, besides messing with ostringstream, is the fact that the printed types are the ones defined on the cpp side rather than what users might be familiar with on the python side:

```
  File "../test_diff_dtype_assert.py", line 7, in <module>
    res1 = torch.addcmul(arg1, arg2, arg3)
RuntimeError: Type promotion unsupported for given input dtypes: Double, Double, ComplexDouble
```

Reply:
other reviewers suggested to remove this, so not going to do it